cloudinit/apport.py

```diff
@@ -155,6 +156,25 @@ def attach_installer_files(report, ui=None):
         attach_file_if_exists(report, realpath, apport_file.label)
 
 
+def attach_ubuntu_pro_info(report, ui=None):
+    """Attach ubuntu pro logs and tag if keys present in user-data."""
+    init = stages.Init()
+    has_up_key = init.cfg and (
+        "ubuntu_advantage" in init.cfg.keys()
+        or "ubuntu-advantage" in init.cfg.keys()
+    )
+    if not has_up_key:
+        return
+
+    realpath = os.path.realpath("/var/log/ubuntu-advantage.log")
+    attach_file_if_exists(report, realpath)
```

Comment:
Any reason why we wouldn't just want to include this log if it exists?
If we only want to conditionally include the ubuntu-pro tag, we could check
if '.var.log.ubuntu-avantage.log' in report.keys():
    report.setdefault("Tags", "")
     if report["Tags"]:
         report["Tags"] += " "
     report["Tags"] += "ubuntu-pro"



Alternatively, if we want to check for ubuntu-advantage or ubuntu_advantage keys in user-data and only include ubuntu-advantage.logs in that case... we probably should read `/run/cloud-init/combined-cloud-config.json`  instead of stages.Init() which has to re-read and merge all system config, user-data and vendor-data.

Reply:
Thanks, @blackboxsw. The presence of the log file is a good enough indicator of 'pro has run'. I have simplified the code.